langserver/symbol.go

```diff
@@ -488,3 +479,91 @@ var symbolCacheTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
 func init() {
 	prometheus.MustRegister(symbolCacheTotal)
 }
+
+func listPkgs(ctxt *build.Context, dir string) []string {
```

Comment:
Perhaps rename to `expandPattern`, and document with what you put in your PR description:

> This code is essentially the buildutil.ExpandPattern modified for our usecase. The idea
used can probably be submitted upstream since there are TODOs hinting at wanting this
functionality.

Also, there are other callers of `buildutil.ExpandPatterns` in the langserver, like in `workspace_refs.go` IIRC. If you don't want to tackle that now -- that's OK, just throwing it out there in case you missed it :)